CONTRIBUTING.md

```diff
@@ -0,0 +1,60 @@
+- **DO** keep the discussions around contributions focused. If you have another matter to discuss, rather than creating a massive tangent in the current discussion, open up a new one.
+- **DO** follow the code of conduct if discussing on GitHub and/or the Discord #rules if discussing on Discord.
+- **DON'T** make PRs that don't directly affect the end user, such as style changes. These are best done as part of a PR related to the area in question. Documentation is fine (and encouraged!), as this is useful to the end user.
+- **DON'T** commit code you didn't write without following its license and following our guidelines for doing so in the Contributor License Agreement. If you are unable to license some code, don't commit it.
+- **DON'T** surprise us with big pull requests or big API changes without talking to us first!
+- **DON'T** make PRs for legal or administrative documents, such as the license, file headers, or code of conduct. If you see something off, let us know and we'll change it.
+
+## Breaking changes
+Contributions must maintain API signature and behavioral compatibility. Contributions that include breaking changes will be rejected. There are exceptions to this, such as if the contribution is made against a major/breaking version branch (such as `2.0` instead of `master`), however this is subject to the team's approval. At risk of sounding like a broken record, talk to us about your idea first!
+
+## Suggested workflow
+Silk.NET uses and encourages [Early Pull Requests](https://medium.com/practical-blend/pull-request-first-f6bb667a9b6). Please don't wait until you're done to open a PR!
+
+1. Install [Git](https://git-scm.com/downloads) and the [.NET Core SDK](https://www.microsoft.com/net/download)
+1. [Fork Silk.NET](https://github.com/Ultz/Silk.NET/fork)
+1. Create a branch on your fork.
+1. Add an empty commit to start your work off (and let you open a PR): `git commit --allow-empty -m "start of [thing you're working on]"`
+1. Open a Pull request with `[WIP]` in the title. Do this **before** you actually start working.
+1. Make your commits in small, incremental steps with clear descriptions.
+1. Tag a maintainer when you're done and ask for a review!
+
+## "Help wanted" & "Good first issue"
+If the core team are unable to champion an issue, they will mark it with a "help wanted" label. This indicates that any external contributor may pick up on this issue and implement it as part of a contribution. However, some "help wanted" issues may require intermediate knowledge of the codebase, area, and/or ecosystem; or may have uncertainty surrounding implementation details (if this is the case talk to us in the Discord or in a discussion issue). Because of this, we also mark some issues with the label "good first issue" which indicates that an issue is straightforward, and is a good place to start if you're interested in contributing but new to the codebase.
```

Comment:
Imo there isn't really any point in using the help wanted label. Any issue that doesn't already have a tracking issue can be worked on, right?
We could look into having a separate label that is between intermediate and good first issue

Reply:
The help wanted label is really just "the core silk.net team won't do this without the community's help", whereas most of the other issues i'm sure we will pick up on in time.

Besides, "help wanted" and "good first issue" are recognised by github and pop up in the "pro tips" section when navigating our issues.

Reply:
I see, welp then it's fine I guess